src/Session.cc

```diff
@@ -575,9 +585,16 @@ const AddressSpace::Mapping Session::recreate_shared_mmap(
     new_map = remote.task()->vm()->mapping_of(new_addr);
     if (preserved_data) {
       memcpy(new_map.local_addr, preserved_data, size);
-      munmap(preserved_data, size);
     }
   }
+  // A MAP_FIXED | MAP_SHARED mmap was created in [m.map.start(),
+  // m.map.end()) above in create_shared_mmap() so the Mapping
+  // maybe_detach_mapping.local_addr must be unmapped. Even if the
+  // create_shared_mmap was unsuccessful, now that the mapping is detached, to
+  // prevent a memory leak, it must still be unmapped
+  if (maybe_local_unmap) {
```

Comment:
The comment is a bit confusing. Just delete it.

Reply:
Done